src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -440,7 +481,11 @@ public void DoWriteIfNeeded()
                 }, this);
 
                 Self._head = _lockedEnd.Block;
-                Self._head.Start = _lockedEnd.Index;
```

Comment:
All I can think is the `Write` callback actually completes before the next lines are run? Was going to move the set above the `Write` but didn't feel comfortable enough with the code.